docs/src/architecture/09_architecture_decisions/0010-frontend-json-schema-constraints.md

```diff
@@ -0,0 +1,170 @@
+| Performance degrades with complex schemas | App becomes slow when rendering large documents | Implement performance testing for templates and optimize lookup algorithms |
+
+## Testing Strategy
+
+To ensure the integrity of our approach:
+
+1. **Unit Tests** - Verify nodeId resolution logic works correctly.
+2. **Integration Tests** - Check that UI components correctly display data from schema nodes.
+3. **Schema Validation** - Create validators that check for required nodeIds in templates.
+4. **Migration Tests** - Verify that document migrations preserve required fields.
+5. **Error Handling Tests** - Verify that the system gracefully handles missing or malformed nodes.
+6. **Performance Tests** (*Optional*) - Measure rendering time with different schema complexities.
+
+## Maintenance Plan
+
+### Current Issues
+
+The frontend team currently spends an excessive amount of time reviewing schema change PRs from architects
+who often make changes without validating against frontend constraints.
+This creates an unsustainable workflow where:
+
+1. Contributors create schema changes without understanding frontend requirements
+2. PRs reach review stage with fundamental issues that break nodeId lookups
+3. Frontend team must thoroughly review every schema change
+4. Cycles of revisions delay implementation and waste development resources
+
+### Improved Process
+
+To address these issues, we suggest implementing the following improved maintenance process:
+
+#### Automated Validation
+
+* **CI/CD Integration**: Add an automated check in the CI pipeline that fails PRs with breaking schema changes
+* **Schema Validator Tool** (*Optional*): Develop a command-line tool that validates schema changes against nodeId requirements
+* **Pre-commit Hook** (*Optional*): Provide a pre-commit hook that architects can install to validate locally before pushing
+
+#### Schema Change Workflow
+
+1. **Intent Documentation**: Architects must document intended schema changes in a standardized format.
+2. **Pre-validation**: Before creating a PR, contributors must run the validation tool against their changes.
+3. **Breaking Change Protocol**: If a breaking change is necessary, architects must:
+   * Document the breaking change
+   * Provide a migration path
+   * Coordinate with the frontend team before submission
+4. **Automated Tests**: Add test cases for each predefined nodeId that verify it still works after schema changes
+
+#### Clear Responsibilities
+
+* **Schema Owners**: Designate specific team members as "schema owners" who understand both architectural and frontend requirements.
+* **Review Rotation**: Establish a rotation of frontend developers responsible for schema reviews to distribute the burden.
+* **Knowledge Transfer**: Conduct regular knowledge sharing sessions on schema constraints for architects.
```

Comment:
Tghis assumes teh architects are responsible for all time for the proposal template schemas.  They are not.  this has been discussed AT LENGTH.  Had anyone paid attention.  Proposal Templates will come ultimately from the Proposal builder in the FRONT END.  And it will conform to the Meta Template requirements.
So this process is excessive given it immediately vanishes when the front end can actually implement the proposal builder functionality as originally intended.